File: FCTrie.java
/* Names: Erin Cole and Christina Cook
 * Filename: FCTrie.java
 * Description: This class is responsible for building the trie that connects
 * 		the fortune databases.
 */

public class FCTrie {
     private FCNode root;

     public FCTrie(FCNode root) {
          root = new FCNode();
     }
     public boolean insert(String[] children) {
          return false;
     }
     public boolean find(String[] children) {
          return false;
     }
     //should this be a reference to an array?
     public class FCNode {
          private String[] children;
         
          public FCNode() {} 
     }

}     

File: FCTrie.js
/* Names: Erin Cole and Christina Cook
 * Project Name: Project Fortune Cookie
 * Filename: testFCTrie.js
 * Description: This file tests the FCTrie.js functions
 */

/* Constructor Name: FCTrie
 * Parameters: key
 * Description: this constructor defines the FCTrie
 */
function FCTrie(key) {
	this.key = key;
	this.value;
}

FCTrie.prototype.put = function(name, value) {
	var node = this,
		nemeLength = name.length,
		i = 0,
		currentLetter;
	for (i=0; i < nameLength; i++) {
		currentLetter = name[i];
		node = node[currentLetter] || (node[currentLetter] = new FCTrie(						currentLetter));
	}
	node.value = value;
	node.name = name;
};

FCTrie.prototype.get = function(name) {
	var node = this,
		nameLength = name.length,
		i, node;
	for (i=0; i < nameLength; ++i) {
		if (!(node = node[name[i]])) break;
	}
	return (i === nameLentgh) ? node.value : 'not found';
};

